src/main/java/kitchenpos/products/tobe/application/ProductService.java

```diff
@@ -0,0 +1,81 @@
+package kitchenpos.products.tobe.application;
+
+import kitchenpos.menus.domain.Menu;
+import kitchenpos.menus.domain.MenuProduct;
+import kitchenpos.menus.domain.MenuRepository;
+import kitchenpos.products.domain.Product;
+import kitchenpos.products.domain.ProductRepository;
+import kitchenpos.products.infra.PurgomalumClient;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.UUID;
+
+@Service
+public class ProductService {
+    private final ProductRepository productRepository;
+    private final MenuRepository menuRepository;
+    private final PurgomalumClient purgomalumClient;
+
+    public ProductService(
+        final ProductRepository productRepository,
+        final MenuRepository menuRepository,
+        final PurgomalumClient purgomalumClient
+    ) {
+        this.productRepository = productRepository;
+        this.menuRepository = menuRepository;
+        this.purgomalumClient = purgomalumClient;
+    }
+
+    @Transactional
+    public Product create(final Product request) {
+        final BigDecimal price = request.getPrice();
+        if (Objects.isNull(price) || price.compareTo(BigDecimal.ZERO) < 0) {
+            throw new IllegalArgumentException();
+        }
+        final String name = request.getName();
+        if (Objects.isNull(name) || purgomalumClient.containsProfanity(name)) {
+            throw new IllegalArgumentException();
+        }
+        final Product product = new Product();
+        product.setId(UUID.randomUUID());
+        product.setName(name);
+        product.setPrice(price);
+        return productRepository.save(product);
+    }
+
+    @Transactional
+    public Product changePrice(final UUID productId, final Product request) {
+        final BigDecimal price = request.getPrice();
+        if (Objects.isNull(price) || price.compareTo(BigDecimal.ZERO) < 0) {
+            throw new IllegalArgumentException();
+        }
+        final Product product = productRepository.findById(productId)
+            .orElseThrow(NoSuchElementException::new);
```

Comment:
예외가 발생했을 때, 적절한 에러 메시지를 던져준다면 에러메시지를 통해서 문제의 원인을 파악하기가 더 수월해지지 않을까요? 😃

```suggestion
            .orElseThrow(() -> new NoSuchElementException("상품이 존재하지 않습니다. productId: " + productId));
```